runtime/stdlib/test.go

```diff
@@ -1367,6 +1428,280 @@ var equalMatcherFunction = interpreter.NewUnmeteredHostFunctionValue(
 	},
 )
 
+const beEmptyMatcherFunctionName = "beEmpty"
+
+const beEmptyMatcherFunctionDocString = `
+Returns a matcher that succeeds if the tested value is an array or dictionary,
+and the tested value contains no elements.
+`
+
+var beEmptyMatcherFunctionType = func() *sema.FunctionType {
+	return &sema.FunctionType{
+		IsConstructor:        false,
+		TypeParameters:       []*sema.TypeParameter{},
+		Parameters:           []sema.Parameter{},
+		ReturnTypeAnnotation: sema.NewTypeAnnotation(matcherType),
+	}
+}()
+
+var beEmptyMatcherFunction = interpreter.NewUnmeteredHostFunctionValue(
+	beEmptyMatcherFunctionType,
+	func(invocation interpreter.Invocation) interpreter.Value {
+		beEmptyTestFunc := interpreter.NewHostFunctionValue(
+			nil,
+			matcherTestFunctionType,
+			func(invocation interpreter.Invocation) interpreter.Value {
+				var isEmpty bool
+				switch value := invocation.Arguments[0].(type) {
+				case *interpreter.ArrayValue:
+					isEmpty = value.Count() == 0
+				case *interpreter.DictionaryValue:
+					isEmpty = value.Count() == 0
+				default:
+					panic(errors.NewUnexpectedError("expected Array or Dictionary argument"))
```

Comment:
"Unexpected" for errors means that an *implementation* error occurred, i.e. execution should have never reached this place, and if so, that indicates a bug in the Cadence implementation itself.

However, in this case here, it is possible that this place is reached, as the static function type is `AnyStruct`, and so any value other than an array or dictionary will reach this default case.

This is actually a *user* error, i.e. the user used the function in a wrong way, which indicates that there is a bug in the Cadence program being executed.

```suggestion
					panic(errors.NewDefaultUserError("expected Array or Dictionary argument"))
```

The other panics below should also be changed to user errors.

Reply:
Thanks for clarifying this part :pray: 
I have updated the panic errors in https://github.com/onflow/cadence/pull/2420/commits/3428082350091c7f1dd0caddd98689332ba05ab1.